tfhe/src/shortint/engine/mod.rs

```diff
@@ -114,6 +114,9 @@ where
             .iter_mut()
             .for_each(|a| {
                 let f_eval = f(i as u64);
+
+                assert!(f_eval < modulus_sup as u64);
+
```

Comment:
so same thing having a bit on the padding bit is not illegal per se, but we may want to return the information as an Enum/Result down the line as this may lead to invalid computations later, don't know what the best here would be

@tmontaigu if you have an opinion on this ?

also I wonder if it may not break some stuff like Parmesan by Jakub which relied on some tricks to have negacyclic functions

Reply:
There is a place in the integers comparision where we do a PBS on values where the bit of padding could be 1